src/ucs/memory/rcache.c

```diff
@@ -746,6 +747,13 @@ void ucs_rcache_check_inv_queue_slow(ucs_rcache_t *rcache)
     pthread_rwlock_unlock(&rcache->lock);
 }
 
+void ucs_rcache_query(ucs_rcache_t *rcache, ucs_rcache_attr_t *rcache_attr)
+{
+    rcache_attr->num_regions   = rcache->num_regions;
+    rcache_attr->total_size    = rcache->total_size;
+    rcache_attr->num_evictions = rcache->num_evictions;
```

Comment:
maybe define `ucs_rcache_attr_t` inside `rcache` for counters and do `memcpy` here?

Reply:
i'd prefer word-by-word copy so it will be atomic (to read rcache fields without a lock and get a self-consistent value)